jvector-base/src/main/java/io/github/jbellis/jvector/graph/GraphIndexBuilder.java

```diff
@@ -559,13 +562,6 @@ public synchronized long removeDeletedNodes() {
         if (nRemoved == 0) {
             return 0;
         }
-        // make a list of remaining live nodes
```

Comment:
As I was looking in the PQ issue, I found that our tests might not cover a situation of missing connectivity in the resulting graph.
Might want to add a test for that to verify that aside of recall being the same for predefined parameters our graph remains fully traversable by beam search.

Reply:
Ensuring navigability in practice is quite tricky. I understand the rationale behind this request, but practical indices are usually not fully navigable anyway.